src/pip/_internal/vcs/__init__.py

```diff
@@ -228,14 +228,15 @@ def get_url_rev_and_auth(self, url):
 
         Returns: (url, rev, (username, password)).
         """
-        error_message = (
-            "Sorry, '%s' is a malformed VCS url. "
-            "The format is <vcs>+<protocol>://<url>, "
-            "e.g. svn+http://myrepo/svn/MyApp#egg=MyApp"
-        )
-        assert '+' in url, error_message % url
-        url = url.split('+', 1)[1]
         scheme, netloc, path, query, frag = urllib_parse.urlsplit(url)
+        if '+' not in scheme:
+            raise AssertionError(
```

Comment:
Should this raise a different error; instead of AssertionError here?

Reply:
I agreee, I'd go with a ValueError (or some custom one).